### PR TITLE
Mass-media console update

### PR DESCRIPTION
**Pull request by MishaUnity**
## About the PR

**A few minor changes:**

- Command access allows to delete any articles, regardless of who the author is.
- Added sound, and popup, when deletion is denied
- Updated mass-media console interface

Most likely, the latest news update...

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image_1

**Changelog**

:cl: Misha_Unity
- tweak: Command access allows to delete any articles
- tweak: Updated mass-media console interface

### CODE REVIEW

File: Content.Server/MassMedia/Systems/NewsSystem.cs [hunk 1]
```diff
@@ -131,6 +134,8 @@ public void OnWriteUiShareMessage(EntityUid uid, NewsWriteComponent component, N
             }
         }
 
+        _audio.PlayPvs(_audio.GetSound(component.ConfirmSound), uid);
```
Comment (ficcialfaint) [inline Content.Server/MassMedia/Systems/NewsSystem.cs:137]: I'm pretty sure you can just use `component.ConfirmSound` without using `GetSound` method here

File: Content.Server/MassMedia/Systems/NewsSystem.cs [hunk 2]
```diff
@@ -146,20 +151,15 @@ public void OnWriteUiDeleteMessage(EntityUid uid, NewsWriteComponent component,
         if (msg.ArticleNum > _articles.Count)
             return;
 
-        var articleToDelete = _articles[msg.ArticleNum];
-        if (articleToDelete.AuthorStationRecordKeyIds == null || !articleToDelete.AuthorStationRecordKeyIds.Any())
+        if (CheckDeleteAccess(_articles[msg.ArticleNum], uid, msg.Session.AttachedEntity))
         {
             _articles.RemoveAt(msg.ArticleNum);
+            _audio.PlayPvs(_audio.GetSound(component.ConfirmSound), uid);
```
Comment (ficcialfaint) [inline Content.Server/MassMedia/Systems/NewsSystem.cs:157]: see comment above

File: Content.Server/MassMedia/Systems/NewsSystem.cs [hunk 3]
```diff
@@ -146,20 +151,15 @@ public void OnWriteUiDeleteMessage(EntityUid uid, NewsWriteComponent component,
...
         {
-            var author = msg.Session.AttachedEntity;
-            if (author.HasValue
-                && _accessReader.FindStationRecordKeys(author.Value, out var recordKeys)
-                && recordKeys.Intersect(articleToDelete.AuthorStationRecordKeyIds).Any())
-            {
-                _articles.RemoveAt(msg.ArticleNum);
-            }
+            _popup.PopupEntity(Loc.GetString("news-write-no-access-popup"), uid);
+            _audio.PlayPvs(_audio.GetSound(component.NoAccessSound), uid);
```
Comment (ficcialfaint) [inline Content.Server/MassMedia/Systems/NewsSystem.cs:162]: see comment(s) above

File: Content.Server/MassMedia/Systems/NewsSystem.cs [hunk 4]
```diff
@@ -217,6 +217,32 @@ private void UpdateWriteDevices()
...
+        }
+        else
+        {
+            if (user.HasValue
+                && _accessReader.FindStationRecordKeys(user.Value, out var recordKeys)
+                && recordKeys.Intersect(articleToDelete.AuthorStationRecordKeyIds).Any())
+            {
+                return true;
+            }
+        }
```
Comment (ficcialfaint) [inline Content.Server/MassMedia/Systems/NewsSystem.cs:241]: You don't need that `else` here

File: Content.Server/MassMedia/Systems/NewsSystem.cs [hunk 5]
```diff
@@ -217,6 +217,32 @@ private void UpdateWriteDevices()
...
+            user.HasValue &&
+            _accessReader.IsAllowed(user.Value, accessReader))
+        {
+            return true;
+        }
+
+        if (articleToDelete.AuthorStationRecordKeyIds == null || !articleToDelete.AuthorStationRecordKeyIds.Any())
+        {
+            return true;
+        }
```
Comment (ficcialfaint) [inline Content.Server/MassMedia/Systems/NewsSystem.cs:232]:
```suggestion
        if (articleToDelete.AuthorStationRecordKeyIds == null || !articleToDelete.AuthorStationRecordKeyIds.Any())
            return true;
```

Comment (MishaUnity) [reply Content.Server/MassMedia/Systems/NewsSystem.cs]: So it reads better with { }, i think...

Comment (ficcialfaint) [reply Content.Server/MassMedia/Systems/NewsSystem.cs]: A'ight

**Review by ficcialfaint — APPROVED** [review]

**Comment by potato1234x** [comment]
censorship :trollface:

**Pull request merged**